documentation/add/04_mpi_framework.md

```diff
@@ -1,71 +1,108 @@
+A standard message passing framework consists of a communicator part, which controls parallel processes and 
+an information exchange medium, responsible for point-to-point or collective interaction between these processes.
 
 To accommodate different demands of different users on different systems, Horace can utilize three different **message transfer media**, namely:
 
-1. Communications via message files written/read from a HDD,
+1. Communications via message files written/read from a storage device,
 2. Using Matlab parallel computing toolbox MPI implementation (based on the toolbox lab`**` operations) and finally,
-3. Communication using standard MPI framework. Currently used MPI implementations are Microsoft MPI for Windows and MPICH for Linux.
+3. Communication using standard MPI framework. Currently supported MPI implementations are Microsoft MPI for Windows and MPICH for Linux.
  
-The **communicator/controller** [[1]](https://en.wikipedia.org/wiki/Message_Passing_Interface#Communicator) parts for these media, responsible for controlling the parallel processes, are correspondingly:
+The **communicator/controller** [[1]](https://en.wikipedia.org/wiki/Message_Passing_Interface#Communicator) parts for these media, 
+responsible for controlling the parallel processes, are correspondingly:
 
 1. Pool of Matlab sessions or compiled Matlab sessions, launched by Matlab's Java launcher,
 2. Matlab Parallel computing toolbox job control mechanism and
 3. **mpiexec** launcher, directly controlling standard MPI processes.
  
-Additional controller will be necessary to use Horace parallelization on a public cluster. This controller wraps around and interfaces the cluster job submission mechanism (e.g. *qsub* or *bsub*)
+An additional controller will be necessary to use Horace parallelization on a public cluster. 
+This controller wraps around and interfaces the cluster job submission mechanism (e.g. *qsub* or *bsub*).
 
-The option 1) suits the users who do not want to compile C\++ code on a Unix system and do not have Matlab parallel computing toolbox installed, option 2) is best for the people, who has Matlab parallel computing toolbox and 3) -- for experienced users who can compile C++ code and set up MPI framework.
+Option 1) suits the users who do not want to compile C\++ code and do not have Matlab parallel computing toolbox installed.
 
-To provide simple switching between different frameworks, the common wrappers are written around the libraries and programs controlling parallel processes and transferring the messages. The wrappers provide common interface to users job. A Horace parallel job tasks are controlled and communicate with each other using this interface, which makes them oblivious to the actual mechanism used for the communications.
+Option 2) is best for the people, who has Matlab parallel computing toolbox.
+
+Option 3) is best for for experienced users who can compile C++ code and set up MPI framework.
```

Comment:
```suggestion
Option 3) is best for users who have acess to Herbert's compiled C++ binaries, and do not have access to Matlab's Parallel Computing Toolbox.
```

Realize this wasn't originally written by you, but users shouldn't need to compile anything to run our C++ framework. We package the compiled binaries they need.